src/EditorFeatures/CSharpTest/SplitOrMergeIfStatements/MergeConsecutiveIfStatementsTests.cs

```diff
@@ -0,0 +1,1685 @@
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        else [|if (|]b)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedOnElseIfOverreachingSelection2()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        else [|if (b)
+        |]{
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedOnElseIfOverreachingSelection3()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        [|else if (b)
+        {|]
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedOnElseIfConditionSelection1()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        else if ([|b|])
+        {
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedOnElseIfConditionSelection2()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        else if [|(b)|]
+        {
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedOnElseIfBodySelection()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        else if (b)
+        [|{
+        }|]
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedOnElseIfConditionCaret()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        else if ([||]b)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedOnElseIfBodyCaret1()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        else if (b)
+        [||]{
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedOnElseIfBodyCaret2()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        else if (b)
+        {
+        }[||]
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedOnParentIf()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        [||]if (a)
+        {
+        }
+        else if (b)
+        {
+        }
+    }
+}");
+        }
```

Comment:
nit: consider extracting all the 'selection oriented' tests into their own partial type.

Reply:
I changed the selection oriented tests to use Theory in ad9208397c765488be8d2e0bee850e33903b9855, which significantly reduced their numbers. Do you think that's good enough or should I still put them in separate classes?

Reply:
the theory approach works great. thanks!